src/main/webapp/app/admin/legal/legal-document-update.component.ts

```diff
@@ -0,0 +1,118 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { faBan, faCheckCircle, faCircleNotch, faExclamationTriangle, faSave } from '@fortawesome/free-solid-svg-icons';
+import { LegalDocumentService } from 'app/shared/service/legal-document.service';
+import { MarkdownEditorComponent, MarkdownEditorHeight } from 'app/shared/markdown-editor/markdown-editor.component';
+import { NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
+import { UnsavedChangesWarningComponent } from 'app/admin/legal/unsaved-changes-warning/unsaved-changes-warning.component';
+import { LegalDocument, LegalDocumentLanguage, LegalDocumentType } from 'app/entities/legal-document.model';
+import { ActivatedRoute } from '@angular/router';
+import { Observable, tap } from 'rxjs';
+import { JhiLanguageHelper } from 'app/core/language/language.helper';
+
+@Component({
+    selector: 'jhi-privacy-statement-update-component',
+    styleUrls: ['./legal-document-update.component.scss'],
+    templateUrl: './legal-document-update.component.html',
+})
+export class LegalDocumentUpdateComponent implements OnInit {
+    legalDocument: LegalDocument;
+    legalDocumentType: LegalDocumentType = LegalDocumentType.PRIVACY_STATEMENT;
+    supportedLanguages: LegalDocumentLanguage[] = [LegalDocumentLanguage.GERMAN, LegalDocumentLanguage.ENGLISH];
+    unsavedChanges = false;
+    faBan = faBan;
+    faSave = faSave;
+    faExclamationTriangle = faExclamationTriangle;
+    faCheckCircle = faCheckCircle;
+    faCircleNotch = faCircleNotch;
```

Comment:
```suggestion
    readonly supportedLanguages: LegalDocumentLanguage[] = [LegalDocumentLanguage.GERMAN, LegalDocumentLanguage.ENGLISH];
    unsavedChanges = false;
    readonly faBan = faBan;
    readonly faSave = faSave;
    readonly faExclamationTriangle = faExclamationTriangle;
    readonly faCheckCircle = faCheckCircle;
    readonly faCircleNotch = faCircleNotch;
```
nitpick: consistent `readonly` marker

potential addition: sorting the attributes into readonly constants first, then an empty line, and then the changeable ones would make it easier to see what is defined at least for me.